### PR TITLE
flipped tracing directionality

### PR DESCRIPTION
**Pull request by PietroPasotti**
Tempo becomes provider of tracing.

Tandem PRs:
- https://github.com/canonical/charm-relation-interfaces/pull/92
- https://github.com/canonical/traefik-k8s-operator/pull/207
- https://github.com/canonical/prometheus-k8s-operator/pull/504
- kratos?

### CODE REVIEW

**Comment by PietroPasotti** [comment]
@shipperizer keep an eye on this please
dangers of early adoption :)

**Comment by shipperizer** [comment]
is it backwards incompatible?ETA for the merge?
anyway got the tickets added in the backlog and we'll tackle this next sprint (from monday) 

thanks for the heads up @PietroPasotti

**Review by sed-i — APPROVED** [review]

**Comment by PietroPasotti** [comment]
> is it backwards incompatible?ETA for the merge? anyway got the tickets added in the backlog and we'll tackle this next sprint (from monday)
> 
> thanks for the heads up @PietroPasotti

Starting tomorrow I'm off until September, so you'll need to coordinate with @simskij and @sed-i on how to handle this, probably you should synchronize with this, prometheus and traefik else the cos-lite edge deployment will be briefly degraded (but hopefully it won't explode)

**Comment by PietroPasotti** [comment]
holding off the merge for now given the above

**Comment by bencekov** [comment]
I'm having trouble packing the charm on this branch. I get this error:
```
ERROR: Double requirement given: opentelemetry-exporter-otlp-proto-grpc (from -r requirements.txt (line 10)) (already in opentelemetry-exporter-otlp-proto-grpc==1.17.0, name='opentelemetry-exporter-otlp-proto-grpc')
```
I checked the requirements files, but to me it doesn't seem like there are multiple imports.

Anyone else has issues like this?

File: tests/interface/test_tracing.py [hunk 1]
```diff
@@ -7,9 +7,7 @@ def test_tracing_v0_interface(interface_tester: InterfaceTester):
     interface_tester.configure(
         interface_name="tracing",
         interface_version=0,
-        # todo replace with upstream CRI when
-        #  https://github.com/canonical/charm-relation-interfaces/pull/41 lands
-        repo="https://github.com/PietroPasotti/charm-relation-interfaces",
-        branch="tracing",
+        repo="https://github.com/canonical/charm-relation-interfaces",
+        branch="main",
```
Comment (sed-i) [inline tests/interface/test_tracing.py:11]: Interface test is failing now:
```
ERROR    interface_tester.interface_test:runner.py:55 Failed check 3: validating schema on scene output: 2 validation errors for ProviderSchema
app -> url
  field required (type=value_error.missing)
app -> ingesters -> 0 -> protocol
  field required (type=value_error.missing)
```
In https://github.com/canonical/charm-relation-interfaces/pull/92, the filename `interfaces/tracing/v0/interface_tests/requirer_tests.py` wasn't renamed to `proiver_tests`. Does it matter?

@simskij @PietroPasotti

**Pull request merged**